cfme/cloud/security_group.py

```diff
@@ -0,0 +1,264 @@
+# -*- coding: utf-8 -*-
+import attr
+
+from cfme.base.ui import BaseLoggedInPage
+from cfme.exceptions import ItemNotFound, SecurityGroupsNotFound
+from cfme.modeling.base import BaseCollection, BaseEntity
+from cfme.utils.appliance.implementations.ui import CFMENavigateStep, navigate_to, navigator
+from cfme.utils.wait import wait_for
+from navmazing import NavigateToAttribute, NavigateToSibling
+from widgetastic_manageiq import (
+    Accordion, BaseEntitiesView, BaseListEntity, BaseQuadIconEntity, BaseTileIconEntity,
+    BootstrapSelect, BreadCrumb, ItemsToolBarViewSelector, ManageIQTree, NonJSBaseEntity,
+    ParametrizedView, SummaryTable, Text, TextInput)
+from widgetastic_patternfly import Button, Dropdown, FlashMessages
+from widgetastic.widget import Select, View
+
+
+class SecurityGroupToolbar(View):
+    configuration = Dropdown('Configuration')
+    policy = Dropdown('Policy')
+    download = Dropdown('Download')
+
+    view_selector = View.nested(ItemsToolBarViewSelector)
+
+
+class SecurityGroupDetailsToolbar(View):
+    configuration = Dropdown('Configuration')
+    policy = Dropdown('Policy')
+    download = Button(title='Download summary in PDF format')
+
+
+class SecurityGroupDetailsAccordion(View):
+    @View.nested
+    class properties(Accordion):  # noqa
+        tree = ManageIQTree()
+
+    @View.nested
+    class relationships(Accordion):  # noqa
+        tree = ManageIQTree()
+
+
+class SecurityGroupQuadIconEntity(BaseQuadIconEntity):
+    pass
+
+
+class SecurityGroupTileIconEntity(BaseTileIconEntity):
+    quad_icon = ParametrizedView.nested(SecurityGroupQuadIconEntity)
+
+
+class SecurityGroupListEntity(BaseListEntity):
+    pass
+
+
+class NonJSSecurityGroupEntity(NonJSBaseEntity):
+    quad_entity = SecurityGroupQuadIconEntity
+    list_entity = SecurityGroupListEntity
+    tile_entity = SecurityGroupTileIconEntity
+
+
+class SecurityGroupDetailsEntities(View):
+    breadcrumb = BreadCrumb()
+    title = Text('//div[@id="main-content"]//h1')
+    properties = SummaryTable(title='Properties')
+    relationships = SummaryTable(title='Relationships')
+    smart_management = SummaryTable(title='Smart Management')
+    firewall_rules = SummaryTable(title="Firewall Rules")
+
+
+class SecurityGroupAddEntities(View):
+    breadcrumb = BreadCrumb()
+    title = Text('//div[@id="main-content"]//h1')
+
+
+class SecurityGroupAddForm(View):
+    network_manager = BootstrapSelect(id='ems_id')
+    name = TextInput(name='name')
+    description = TextInput(name='description')
+    cloud_tenant = Select(name='cloud_tenant_id')
+    add = Button('Add')
+    cancel = Button('Cancel')
+
+
+class SecurityGroupView(BaseLoggedInPage):
+    """Base view for header and nav checking, navigatable views should inherit this"""
+
+    flash = FlashMessages('.//div[@id="flash_msg_div"]'
```

Comment:
BaseLoggedInPage already implements a flash widget, does its locator not work on this page?